pkg/api/v1/router.go

```diff
@@ -10,49 +10,99 @@ import (
+				srvAttrs.GET("/:namespace", amw.RequiredScopes(readScopes("server", "server:attributes")), r.serverAttributesGet)
+				srvAttrs.PUT("/:namespace", amw.RequiredScopes(updateScopes("server", "server:attributes")), r.serverAttributesUpdate)
+				srvAttrs.DELETE("/:namespace", amw.RequiredScopes(deleteScopes("server", "server:attributes")), r.serverAttributesDelete)
+			}
+
+			// /servers/:uuid/components
+			srvComponents := srv.Group("/components")
+			{
+				srvComponents.POST("", amw.RequiredScopes(createScopes("server", "server:component")), r.serverComponentsCreate)
+				srvComponents.GET("", amw.RequiredScopes(readScopes("server", "server:component")), r.serverComponentGet)
+				srvComponents.PUT("", amw.RequiredScopes(updateScopes("server", "server:component")), r.serverComponentUpdate)
+			}
+
+			// /servers/:uuid/secrets/:slug
+			svrSecret := srv.Group("secrets/:slug")
+			{
+				svrSecret.GET("", amw.RequiredScopes([]string{"read:server:secrets"}), r.serverSecretGet)
+				svrSecret.PUT("", amw.RequiredScopes([]string{"write:server:secrets"}), r.serverSecretUpsert)
+				svrSecret.DELETE("", amw.RequiredScopes([]string{"write:server:secrets"}), r.serverSecretDelete)
+			}
+
+			// /servers/:uuid/versioned-attributes
+			srvVerAttrs := srv.Group("/versioned-attributes")
+			{
+				srvVerAttrs.GET("", amw.RequiredScopes(readScopes("server", "server:versioned-attributes")), r.serverVersionedAttributesList)
+				srvVerAttrs.POST("", amw.RequiredScopes(createScopes("server", "server:versioned-attributes")), r.serverVersionedAttributesCreate)
+				srvVerAttrs.GET("/:namespace", amw.RequiredScopes(readScopes("server", "server:versioned-attributes")), r.serverVersionedAttributesGet)
+			}
+		}
+	}
+
+	// /server-component-types
+	srvCmpntType := rg.Group("/server-component-types")
+	{
+		srvCmpntType.GET("", amw.RequiredScopes(readScopes("server-component-types")), r.serverComponentTypeList)
+		srvCmpntType.POST("", amw.RequiredScopes(updateScopes("server-component-types")), r.serverComponentTypeCreate)
+	}
+
+	// /server-component-firmwares
+	srvCmpntFw := rg.Group("/server-component-firmwares")
+	{
+		srvCmpntFw.GET("", amw.RequiredScopes(readScopes("server")), r.serverComponentFirmwareList)
+		srvCmpntFw.POST("", amw.RequiredScopes(createScopes("server")), r.serverComponentFirmwareCreate)
+		srvCmpntFw.GET("/:uuid", amw.RequiredScopes(readScopes("server")), r.serverComponentFirmwareGet)
+		srvCmpntFw.PUT("/:uuid", amw.RequiredScopes(updateScopes("server")), r.serverComponentFirmwareUpdate)
+		srvCmpntFw.DELETE("/:uuid", amw.RequiredScopes(deleteScopes("server")), r.serverComponentFirmwareDelete)
+	}
+
+	// /server-secret-types
+	srvSecretTypes := rg.Group("/server-secret-types")
+	{
+		srvSecretTypes.GET("", amw.RequiredScopes(readScopes("server-secret-types")), r.serverSecretTypesList)
+		srvSecretTypes.POST("", amw.RequiredScopes(createScopes("server-secret-types")), r.serverSecretTypesCreate)
```

Comment:
I'm not super sure about this question: Should we have separate scopes for administrating secret types? I can see this being a super-admin or a dedicated client.

Reply:
This uses the standard readScopes and createScopes functions which allow the admin scopes to still apply. Not sure if that is good enough for what you were thinking on this or not.

Reply:
so, my suggestion was to remove those scopes and require specific ones instead. Wanted to get some discussion on that.